### ISSUE TITLE
v 0.1.12; stargate mod fills ram

### ISSUE THREAD

**Issue by Schrimpes** [AllTheMods/ATM-9#533]
when I went through the stargate, my ram filled up completely (I give MC 12 GB) and the fps dropped to 3-5. I had to close mc to empty the ram. On the second test to recreate it, MC crashed shortly after the fps drops as I passed through the gate. The mod is pretty nice, but I think it's still too alpha.

btw: 
If you haven't set a home, waypoint or waystone, only the "/spawn" command helps, as I couldn't find an "earth" address on Abydos ^^

![2023-10-07_02 32 46](https://github.com/AllTheMods/ATM-9/assets/93662041/0b42ad50-4c99-458c-9964-d375698c1259)

**Comment by seanmmcd** [comment]
Experiencing this same issue.

**Comment by Alucard1313** [comment]
Same issue

**Comment by Zarkazm** [comment]
> btw: If you haven't set a home, waypoint or waystone, only the "/spawn" command helps, as I couldn't find an "earth" address on Abydos ^^

On Abydos there are little surface "huts" with Sandstone Slabs on them that have a cartouche inside with a random address.  Sometimes it has the Earth one, sometimes its dimensions that don't have gates yet.

But yeah, I'm crashing on Abydos and Chulak after being there too long as well.  The progression of the mod seems to rely on exploring and finding the cartouches, but crashing every few minutes when you're in those dimensions makes it difficult.  It is actively being worked on at least.  The latest version that came out today re-adds the craftable Stargates from the old SGCraft days.

**Comment by Zarkazm** [comment]
Added the 1.19.2 version of the SGJourney mod to ATM8 to see if it's any different.  (SGJourney dev is updating 1.19.2 and 1.20.1 in parallel, so same mod version).  Also crashes after being on Abydos for a bit, so i'm thinking that mod is indeed quite alpha.

**Comment by tehgreatdoge** [comment]
This might be patched soon. Can't quite tell if this is a duplicate of [this](https://github.com/Povstalec/StargateJourney/issues/21), but it seems like it from the symptoms (no Stellar View installed, sudden fps drop on abydos)

**Comment by Povstalec** [comment]
> This might be patched soon. Can't quite tell if this is a duplicate of [this](https://github.com/Povstalec/StargateJourney/issues/21), but it seems like it from the symptoms (no Stellar View installed, sudden fps drop on abydos)

It is

I've made a hotfix to solve the problem (though I haven't had time to test it properly, so I can't be 100% sure).

The issue most likely lies in the fact that back when I first discovered how to render custom sky, I've made a stupid error by having the sky object be created in the render function, which caused it to be recreated every tick. I've first found out about the error in another one of my mods, Stellar View, which I've since fixed, however I did not check if I've made the same mistake in Stargate Journey (which I have).

The reason it's gone unnoticed up until now is that Stellar View is compatible with Stargate Journey and replaces the default sky renderers on all the planets. I've been using it in the Dev environment and a bunch of other people have been using it alongside Stargate Journey as well, so it effectively hid the problem.
